app/models/submission_completion.rb

```diff
@@ -24,4 +28,10 @@ def perform!
   def create_invoice_for?(submission)
     !submission.report_no_business? && submission.total_spend != 0 && ENV['SUBMIT_INVOICES']
   end
+
+  def mark_old_submission_as_replaced!
+    return unless (old_submission = submission.task.submissions.find_by(aasm_state: 'completed'))
```

Comment:
Is there ever a case where they'd be more than one completed submission on a task? (I'm thinking legacy reset tasks). Would an `order` be useful here?

Reply:
Ah, good thought about the "legacy reset tasks". It has been and will continue to be impossible to create 2 completed submissions against task with the code, but you're right that anything is possible once we're on the console.

We had a look, and on production, there are 16 tasks that have 2 completed submissions. I think we should create a data migration after this has been merged to fix that and mark the older one as replaced.

However, as for this bit of code, either we keep it as it is, or it should mark *all* completed submissions (although there should only ever be one) as replaced. Thoughts?

Reply:
An alternative option, we create a validation on submissions that, if completed, it errors if there's another completed submission on its task. This would prevent anyone from doing dodgy things on the console, even though they should hopefully not have any reason to after replacement submissions is done.

Reply:
I'm leaning towards marking *all* related submissions as replaced - even for those we've previously done something on the console, this makes sense.